core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1280,6 +1309,15 @@ class UnifiedLog(@volatile var logStartOffset: Long,
           case _ => Optional.empty[Integer]()
         }
         Some(new TimestampAndOffset(RecordBatch.NO_TIMESTAMP, logStartOffset, epochOpt))
+      } else if (targetTimestamp == ListOffsetsRequest.EARLIEST_LOCAL_TIMESTAMP) {
+        val offset = _localLogStartOffset
```

Comment:
Change to curlocalLogStartOffset = localLogStartOffset ?

Reply:
Done